build-manual.py

```diff
@@ -73,6 +73,78 @@ def __init__(self, number: str, title: str, filename: str):
 		self.filename = filename
 		self.items = []
 
+def make_one_page(dest_directory):
+	"""
+	Generate one-page php file of the manual.
+	"""
+	# Get all php files in destination directory.
+	php_files = list(filter(lambda x: x.endswith("php"), os.listdir(dest_directory)))
+	php_re = regex.compile(r"<\?.+?\?>", regex.S)
+	index = php_files.pop(php_files.index("index.php"))
+
+	with open(dest_directory + "/" + index, "r", encoding="utf-8") as f:
+		index_soup = BeautifulSoup(f, features="html.parser")
+		php_tags = php_re.findall(index_soup.prettify(formatter=None))
+
+	# The frontmatter contains all needed tags and texts at the beginning of the one-page manual
+	index_soup.find("section")["id"] = "0"
+	frontmatter = index_soup.find("section")
+	# Remove short form ToC
+	frontmatter.find_all("section")[-1].decompose()
+
+	# Get ToC
+	with open(dest_directory + "/" + php_files[0], "r", encoding="utf-8") as f:
+		toc = str(BeautifulSoup(f, features="html.parser").find("nav"))
+
+	# Format hrefs in ToC for onepager
+	toc_re = regex.compile(r"(?<=href\=\")([/a-z0-9\.-]+?)(?=#)")
+	toc = toc_re.sub("", toc)
+	toc_re2 = regex.compile(r"(?<=href\=\")(?P<href>[/0-9a-z-\.]+(?<=/)(?P<chapter>\d+)(?=-)[/0-9a-z-\.]+(?=\"))")
+	toc = toc_re2.sub(r"#\g<chapter>", toc)
+	toc_re3 = regex.compile(r"(?<=href=\")(/manual.+?\d)(?=\")")
+	toc = toc_re3.sub(r"#0", toc)
+
+	# Get chapter tags/text without toc
+	bodymatter = []
+	chapter_re = regex.compile(r"^\d.+")
+	php_files = list(filter(lambda x: chapter_re.match(x), php_files))
+
+	for file in php_files:
+		with open(dest_directory + "/" + file, "r", encoding="utf-8") as f:
+			soup = BeautifulSoup(f, features="html.parser")
+
+		bodymatter.append(soup.find_all("section")[0])
+
+	# Writing the one page manual php file (overwrites if exist)
+	onepage = dest_directory + "/manual-onepage.php"
```

Comment:
We don't really need a variable here if it's just going to be used in one line below